### PR TITLE
DM-22677: Relocate scripts to python module

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: python/lsst/daf/butler/script/dumpButlerConfig.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    if subset is not None:
+        config = config[subset]
+
+    try:
+        config.dump(outfh)
+    except AttributeError:
+        print(config, file=outfh)
+
+
+def main():
```
Comment (TallJimbo) [inline python/lsst/daf/butler/script/dumpButlerConfig.py:89]: Would it be worthwhile having this take an `args` to pass to `parse_args()` that would take `sys.argv[1:]` or similar?  That seems like it's a missing piece of making this more usable, but I also don't really a scenario where we'd actually use that.

File: python/lsst/daf/butler/script/dumpButlerConfig.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    args = build_argparser().parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+
+    try:
+        dumpButlerConfig(args.root, args.searchpath, args.subset, sys.stdout)
+    except Exception as e:
+        print(f"{e}", file=sys.stderr)
```
Comment (TallJimbo) [inline python/lsst/daf/butler/script/dumpButlerConfig.py]: I'm not sure I like this traceback squashing here. In a really mature program that knows about all of its common failure modes, that can give you more friendly error messages, but in code that's under development, so often it just hides what went wrong and forces you to run again, which I find super frustrating.

File: python/lsst/daf/butler/core/repoTransfers.py [hunk 3]
```diff
@@ -165,8 +165,8 @@ def saveDatasets(self, refs: Iterable[DatasetRef], *,
             path generated by the datastore.  If `None`, the `FileDataset`
             returned by `Datastore.export` will be used directly.
 
-        Note
-        ----
+        Notes
+        -----
```
Comment (TallJimbo) [inline python/lsst/daf/butler/core/repoTransfers.py:169]: I think I've also got this fix on a branch for you to review :slightly_smiling_face: .

**Review by TallJimbo — APPROVED** [review]
Two minor comments on one script that probably apply to all of them.

File: python/lsst/daf/butler/script/dumpButlerConfig.py (hunk 1 above)
Comment (timj) [reply python/lsst/daf/butler/script/dumpButlerConfig.py:89]: I believe I'm following standard practice here: https://developer.lsst.io/stack/argparse-script-topic-type.html?highlight=autoprogram

File: python/lsst/daf/butler/script/dumpButlerConfig.py (hunk 2 above)
Comment (timj) [reply python/lsst/daf/butler/script/dumpButlerConfig.py]: I got so wrapped up in ensuring that I returned non-zero exit status that I forgot I could just let it die and also get non-zero exit status.

**Pull request merged**